mmocr/models/textrecog/encoders/restransformer.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
+
+from mmcv.cnn.bricks.transformer import BaseTransformerLayer
+from mmcv.runner import BaseModule, ModuleList
+
+from mmocr.models.builder import BACKBONES
+from mmocr.models.textrecog.backbones.resnet_abi import ResNetABI
+from mmocr.models.textrecog.layers import PositionalEncoding
+
+
+@BACKBONES.register_module()
+class ResTransformer(BaseModule):
+    """Implement ResTransformer backbone for text recognition, modified from
+    `<https://github.com/FangShancheng/ABINet>`.
+
+    Args:
+        base_channels (int): Number of channels of input image tensor.
+        layers (list[int]): List of BasicBlock number for each stage.
+        channels (list[int]): List of out_channels of Conv2d layer.
+        out_indices (None | Sequence[int]): Indices of output stages.
+        last_stage_pool (bool): If True, add `MaxPool2d` layer to last stage.
+    """
+
+    def __init__(self,
+                 n_layers=2,
+                 n_head=8,
+                 d_model=512,
+                 d_inner=2048,
+                 dropout=0.1,
+                 max_len=8 * 32,
+                 init_cfg=None):
```

Comment:
add a parameter for config ResNet:
    cnn_backbone = dict(type=ReaNetABI, ...)
otherwise, it cannot change ResNetABI parameter